### PR TITLE
First pass of 2fa documentation

### PR DESCRIPTION
**Pull request by javorszky**
I've tried to make the descriptions as non-technical as possible. Let me know if you think I need to expand on something / correct others.

### CODE REVIEW

File: 2fa/README.md [hunk 1]
```diff
@@ -2,8 +2,46 @@
 
 ## What is it?
 
+Two factor authentication (2fa from now on) is a method to double check that you're really you by requiring one more check with something you have. This is usually a code generated by an authenticator application or a physical device, or a text message that's sent to your phone.
```
Comment (yorkshiretwist) [inline 2fa/README.md]: Might be worth starting with something well-known. E.g.:

Most people use passwords to access services and information that needs to be kept secure. A password is sometimes called a piece of "something you know" security. However this password could be stolen or guessed by a malicious third party.

Two factor authentication (2FA from now on) is a method to double check that you're really you by requiring additional security information using "something you have". This is usually a code generated by an authenticator application, or a physical device, or a message that's sent to your phone.

For an example imagine you wanted to unlock your car, and it required 2FA. You'd need a password ("something you  know") AND a key ("something you have"). This greatly increases the security of your car.

File: 2fa/README.md [hunk 2]
```diff
@@ -2,8 +2,46 @@
...
+
+When you set up 2fa you provide a special code (seed) of your account / device which will instruct the authenticator to generate the correct codes. This usually happens by scanning a QR code, or by providing the site with a unique identifier of your physical device.
```
Comment (yorkshiretwist) [inline 2fa/README.md]: Are these two paragraphs a bit technical? Full marks for explaining them in clear language, but we might be diving a bit deep here!

File: 2fa/README.md [hunk 3]
```diff
@@ -2,8 +2,46 @@
...
+
+Based on that information, the authenticator will provide a different code that's time based (called a TOTP, or Time based One Time Password). Because only you should have the authenticator set up to your account's seed, other people can't generate the correct code without your device.
+
 ## Why is it important?
 
+Traditionally the way to log into websites, whether it's your bank, Facebook, Twitter, your blog, or any place is to provide a username and password - something that you know. Since there are so many websites now requiring usernames and passwords, and password managers aren't widely used yet, it is a common thing that we reuse passwords across different sites. That also means that if a website got hacked in the past where you used the same username / password as on Facebook, other people could use that information to "guess" your password and gain access to a site that hasn't been compromised yet.
```
Comment (yorkshiretwist) [inline 2fa/README.md]: Bah, I should have read the whole thing before making my comment above. Perhaps my suggestions will stand, but this section will need to be modified a bit.

File: 2fa/README.md [hunk 4]
```diff
@@ -2,8 +2,46 @@
...
+
+With 2fa turned on, these sites will also require the person logging in to provide another code from something that you should have on your person. This means that the security of those accounts are greatly improved as they will most likely not have access to the device that generates the second code, which means that even though they do have your password, they won't be able to log in.
+
 ## What's happening with it?
```
Comment (yorkshiretwist) [inline 2fa/README.md]: Might be worth finding some stats about how many systems do use 2FA (i.e. banks) and what the up-take has been. But, of course, more needs to be done.

**Comment by yorkshiretwist** [comment]
Thanks @javorszky, this is great stuff. I think I commented prematurely as you've covered most of my points later in the article. I would stick with my suggestion that putting the thing most people will be familiar with first (i.e. passwords) will give a better shape to it.

**Comment by yorkshiretwist** [comment]
Hi @javorszky have you though about tweaking the text so it starts with talking about passwords first? I do think that would give a better shape to the article. Let me know if you'd like me to tackle that, Thanks.

**Comment by javorszky** [comment]
Hello!

Yes. Give me until Sunday evening though. The past 2 weeks were conference prep, travels, and currently conference.

**Comment by yorkshiretwist** [comment]
@javorszky Hi Gabor. Have you had chance to look at the suggested changes? If you're too busy at the moment no problem, I can make some time for this. Thanks.

**Pull request merged**